src/useTracking.js

```diff
@@ -0,0 +1,24 @@
+import merge from 'deepmerge';
+import { useContext } from 'react';
+import { ReactTrackingContext } from './withTrackingComponentDecorator';
+
+export default function useTracking() {
+  const trackingContext = useContext(ReactTrackingContext);
+
+  if (!(trackingContext && trackingContext.tracking)) {
+    throw new Error(
+      'Attempting to call `useTracking` ' +
+        'without a ReactTrackingContext present. Did you forget to wrap the top of ' +
+        'your component tree with `track`?'
+    );
+  }
+
+  return {
+    getTrackingData: () => trackingContext.tracking.data,
+    trackEvent: data => {
+      return trackingContext.tracking.dispatch(
+        merge(trackingContext.tracking.data, data)
+      );
+    },
```

Comment:
I think as this is currently written will cause the return value to change on each render (see my note on mixing this with user-land `useEffect` as a potential use case). Would it make sense to `useMemo` or `useRef` here?